### PR TITLE
Add tests for CallVisualizerCoordinator

### PR DESCRIPTION
**Pull request by gersonnoboa**
The delegates inside the coordinators are not tested because each coordinator itself tests the delegate methods functioning through their own set of tests. Also, the coordinators in this class are all private, so testing it from outside would require moving the coordinators to internal properties, which I would say is not desirable.

MOB-2917

**Release notes:**

 - [ ] Feature
 - [x] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [ ] Tests fixed, added? Unit, acceptance, snapshots?
 - [ ] Logging necessary for future troubleshooting of customer issues added?

**Screenshots:**

### CODE REVIEW

File: GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            mobileObservationVideoFps: try videoFps(),
+            mobileObservationEnabled: true
+        )
+
+        coordinator.environment.fetchSiteConfigurations = { callback in
+            callback(.success(site))
+        }
+
+        let answer = Command<Bool> { boolean in
+            XCTAssertTrue(boolean)
```
Comment (igorkravchenko) [inline GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift:51]: This assert will only be called if `answer` command is called. To avoid missing this command call in unit tests I think it would be better to introduce empty array of bools, and append `boolean ` to it and assert it out of the closure like this:

```
var answers: [Bool] = []
let answer = Command<Bool> { boolean in
  answers.append(boolean)
}

XCTAssertEqual(answers, [true])
```

File: GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
```
Comment (igorkravchenko) [inline GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift:84]: Same idea about asserting in callback.

**Review by igorkravchenko — CHANGES_REQUESTED** [review]
Added suggestions about avoiding asserting inside closures.

File: GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift (hunk 1 above)
Comment (gersonnoboa) [reply GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift]: Fixed 21298a0a635189b8bb2191b3dc1b34e4e621e7be

File: GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift (hunk 2 above)
Comment (gersonnoboa) [reply GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift]: I changed it here, because I realized that indeed this assertion wasn't even running, but the problem is that to run it, I need to manually press the AlertViewController because the answer command depends on what the user presses. Thus I believe that because showing the AlertViewController is what the coordinator takes care of, then that should be enough for the test. 21298a0a635189b8bb2191b3dc1b34e4e621e7be let me know if it makes sense.

Comment (igorkravchenko) [reply GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift]: Make sense.

File: GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+        XCTAssertTrue(coordinator.environment.presenter.getInstance()?.presentedViewController is AlertViewController)
+    }
+
+    func test_end() {
+        var isCalled = false
+        
+        coordinator.environment.screenShareHandler.stop = { _ in
+            isCalled = true
+        }
```
Comment (yurii-glia) [inline GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift:96]: would it be beneficial to have a counter here to know about multiple calls if this happens?
```swift
        var stopCallCounter = 0
        
        coordinator.environment.screenShareHandler.stop = { _ in
            stopCallCounter += 1
        }
        // ...
        XCAssertTrue(stopCallCounter)
```

Comment (gersonnoboa) [reply GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift]: Fixed 52f4f8d4ae0dc6303b92c6934efda5240eeaa6e2

**Review by yurii-glia — APPROVED** [review]

**Review by igorkravchenko — APPROVED** [review]

**Comment by gersonnoboa** [comment]
!squash
